### PR TITLE
=str Fold InHandler and OutHandler for operator RetryFlowCoordinator.

### PR DESCRIPTION
**Pull request by He-Pin**
References https://github.com/akka/akka/issues/19752

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/RetryFlowCoordinator.scala [hunk 1]
```diff
@@ -54,9 +54,10 @@ import akka.util.OptionVal
     private var elementInProgress: OptionVal[In] = OptionVal.none
     private var retryNo = 0
 
-    setHandler(
+    setHandlers(
       externalIn,
-      new InHandler {
+      internalOut,
+      new InHandler with OutHandler {
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/RetryFlowCoordinator.scala:60]: shouldn't that even be on the `TimerGraphStageLogic(shape) extends InHandler with OutHandler`?

**Review by patriknw — COMMENTED** [review]
I'm not so sure we should change this. With those 4 handlers it's more difficult to read which belong to in/out. Saving a few allocations for this can't make a difference. Let's see what @johanandren says.

**Review by johanandren — COMMENTED** [review]
Yeah, I agree with @patriknw, I don't think the avoided allocs is it worth making this one harder to follow.

**Comment by patriknw** [comment]
Thanks anyway

**Comment by He-Pin** [comment]
Oops, I forget to close this one:)

**Pull request closed without merging**